src/main/java/com/github/fge/jsonpatch/AddOperation.java

```diff
@@ -136,7 +136,23 @@ private JsonNode addToObject(final JsonPointer path, final JsonNode node)
         final TokenResolver<JsonNode> token = Iterables.getLast(path);
         final JsonNode ret = node.deepCopy();
         final ObjectNode target = (ObjectNode) path.parent().get(ret);
-        target.set(token.getToken().getRaw(), value);
+
+        final String objectName = token.getToken().getRaw();
+        final JsonNode targetObject = target.get(objectName);
+
+        if (targetObject != null && targetObject.isArray()) {
+            addToArrayObject((ArrayNode) targetObject);
+        } else {
+            target.set(objectName, value);
+        }
         return ret;
     }
+
+    private void addToArrayObject(ArrayNode targetObject) {
+        if (value.isArray()) {
```

Comment:
This check is redundant, isn't it? 🤔 You already check that before calling the method, don't you? The else block is dead code then, right?

Reply:
Yes, looks like it. I removed it and tests are still successful 😄